### DISCUSSION TITLE
JZZ.MIDI.Mtrk is not a constructor

### DISCUSSION THREAD

**Discussion by hr30** [jazz-soft/JZZ-midi-SMF#22] [Q&A]
I copied the code from the description of JZZ-midi-SMF into a file jzzTest.js:
```
const JZZ = require('jzz');
require ('jzz-midi-smf')(JZZ);
var smf = new JZZ.MIDI.SMF (0, 24),
    trk = new JZZ.MIDI.SMF.Mtrk (),
    chord = [60, 64, 67, 71];

smf.push (trk);
trk.smfBPM(120).ch(0) // subsequent calls will apply to channel 0
      .tick(96).noteOn('C6', 127).noteOn('Eb6', 127).noteOn('G6', 127)
      .tick(96).noteOff('C6').noteOff('Eb6').noteOff('G6')
      .tick(96).smfEndOfTrack();
```

when starting with `node jzzTest.js` I got the error Message:
```
 trk = new JZZ.MIDI.SMF.Mtrk (),
          ^

TypeError: JZZ.MIDI.SMF.Mtrk is not a constructor
```
I can't catch the fault.

**Comment by jazz-soft** [comment]
Please check your spelling: MTrk

**Reply by hr30** [reply]
Oops ... my fault.
Thanks.

**Comment by hr30** [comment]
Ok, I have corrected that fault. 
Now the program ends without fault, but without sound.
I tried to put in front of that a snippet from another example in that way:
```
JZZ().or('Cannot start MIDI engine!')
    .openMidiOut().or('Cannot open MIDI Out port!')
    .trk. ...
```
but this ends allso in an error message:
```
    .trk.smfBPM(120).ch(0) // subsequent calls will apply to channel 0
        ^

TypeError: Cannot read properties of undefined (reading 'smfBPM')
```
Putting this in an .and-enclosure like `.and (trk. ...)` does not throw an error, but no sound occurs.
Wrapping this into a function like `.and (function () {trk. ...)` throws an error too.

Honestly said, I still have a lack of understanding this concept.

My aim was to use jzz-midi-smf in order to get an infinite sequential play of an array of notes or of random generated notes, and I don't know if this is the right way or how to realize this.

**Comment by jazz-soft** [comment]
trk is  not a member of JZZ().openMidiOut().
I don't understand what you are trying to do.
You can send notes to MIDI Out without jzz-midi-smf.
